File: trunk/SuperShooter/Assets/Scripts/ExplodeOnContact.js
var explosion : GameObject;
var causesExplosion : boolean;
var model : GameObject;
var colideCount = 0;
var collided = false;
function OnCollisionEnter (c : Collision) : void{
	if (!collided && c.contacts.length > 0){
		
		collided = true;
		var contact = c.contacts[0];
		
		
		var collider : Collider = GetComponent(Collider);
		var renderer : Renderer = model.GetComponent(Renderer);
	
		DestroyObject(model);
		collider.isTrigger = true;
		rigidbody.velocity = Vector3.zero;
	
	
		var rot = Random.rotation;
		Instantiate(explosion, contact.point + contact.normal * 0, rot);
		
		
		if (causesExplosion){
			var radius =20.0;
			var power = 5000.0;
			var explosionPos = transform.position;
  	  		var colliders : Collider[] = Physics.OverlapSphere (explosionPos, radius);
    
    		for (var hit in colliders) {
        		if (hit == null)
            		continue;
        		if (!hit.isTrigger && hit.rigidbody){
					hit.gameObject.SendMessage("OnExplosion", contact.point, SendMessageOptions.DontRequireReceiver );
				}
    		} 	  		
			
			colliders = Physics.OverlapSphere (explosionPos, radius);
    
			
    		for (var hit in colliders) {
        		if (hit == null)
            		continue;
        		if (!hit.isTrigger && hit.rigidbody){
					hit.rigidbody.AddExplosionForce(power, explosionPos, radius, 0);
				}
    		}
		}
    
	}
	Invoke ("DestroyNow", 3);
}

function DestroyNow ()
{
	DestroyObject(gameObject);
}

File: SuperShooter/Assets/Scripts/ExplodeOnContact.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class ExplodeOnContact : MonoBehaviour
{
    public GameObject explosion;
    public bool causesExplosion;
    public GameObject model;
    public int colideCount;
    public bool collided;
    public virtual void OnCollisionEnter(Collision c)
    {
        if (!this.collided && (c.contacts.Length > 0))
        {
            this.collided = true;
            ContactPoint contact = c.contacts[0];
            Collider collider = (Collider) this.GetComponent(typeof(Collider));
            Renderer renderer = (Renderer) this.model.GetComponent(typeof(Renderer));
            UnityEngine.Object.DestroyObject(this.model);
            collider.isTrigger = true;
            this.GetComponent<Rigidbody>().velocity = Vector3.zero;
            Quaternion rot = Random.rotation;
            UnityEngine.Object.Instantiate(this.explosion, contact.point + (contact.normal * 0), rot);
            if (this.causesExplosion)
            {
                float radius = 20f;
                float power = 5000f;
                Vector3 explosionPos = this.transform.position;
                Collider[] colliders = Physics.OverlapSphere(explosionPos, radius);
                foreach (Collider hit in colliders)
                {
                    if (hit == null)
                    {
                        continue;
                    }
                    if (!hit.isTrigger && hit.GetComponent<Rigidbody>())
                    {
                        hit.gameObject.SendMessage("OnExplosion", contact.point, SendMessageOptions.DontRequireReceiver);
                    }
                }
                colliders = Physics.OverlapSphere(explosionPos, radius);
                foreach (Collider hit in colliders)
                {
                    if (hit == null)
                    {
                        continue;
                    }
                    if (!hit.isTrigger && hit.GetComponent<Rigidbody>())
                    {
                        hit.GetComponent<Rigidbody>().AddExplosionForce(power, explosionPos, radius, 0);
                    }
                }
            }
        }
        this.Invoke("DestroyNow", 3);
    }

    public virtual void DestroyNow()
    {
        UnityEngine.Object.DestroyObject(this.gameObject);
    }

}